### ISSUE TITLE
OpenAPI URL template variables / values

### ISSUE THREAD

**Issue by prushforth** [opengeospatial/D012-MapML_Engineering_Report#16]
Per a comment on this morning's telecom, I had a look at the [OpenAPI spec for possible guidance on how to specify URL template variable values](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#serverObject).

Although that might be useful for JavaScript programmers, the objective of MapML is to extend the space currently occupied by browser developers, who rely on well-known markup for variable values. So I think the MapML approach of using `<input name="varnamehere" type="...">` is appropriate for the target developer audience.

OTOH, using URL templates *at all* is new in HTML-land, so I can imagine that there will be some discussion about that.  All I can say at this point is that URL templates are common practice in Web maps, and it could probably be beneficial to add it as a concept even to HTML form processing, although I am not an expert in that and we should welcome an expert's perspective on it.

You can close this when / if you add the topic to the ER.

**Comment by joanma747** [comment]
In the example: https://github.com/opengeospatial/WFS_FES/blob/master/openapi-buildings.yaml we can see that the input parametres are defined in an "schema" section using a "yamlizacion" of the json schema vocabulary. This mean that you can defined the data type (string, number...) and some constraints: maximum, minimum and enums, as well as default values. like in this example.
```yaml
 default: residential
 type: string
        enum:
          - residential
          - commercial
          - public use
```
The use of `<select>` and `<option selected>' allows the same functionality. `<input type="range">` allows for max and min values. This way we can have an homogeneous notation without having to introduce another encoding. The good thing of yaml is that you can specify object types that will not be possible in a html form easily.

In conclusion we will use an element link with a tref with a URL template that has child elements ``<input>' and  `<select>` with the same "name" of the variables of the URL template, specifying the possible values of the variables.

If we agree with this, we can close the issue.
